pkg/blockdeviceclaim/v1alpha1/build.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	ndmapisv1alpha1 "github.com/openebs/maya/pkg/apis/openebs.io/ndm/v1alpha1"
+)
+
+//TODO: While using these packages UnitTest must be written to corresponding function
+
+// Builder is the builder object for BlockDeviceClaim
+type Builder struct {
+	BDC  *BlockDeviceClaim
+	errs []error
+}
+
+// NewBuilder returns an empty instance of the Builder object
+func NewBuilder() *Builder {
+	return &Builder{
+		BDC: &BlockDeviceClaim{&ndmapisv1alpha1.BlockDeviceClaim{}},
+	}
+}
+
+// BuilderForObject returns an instance of the Builder object based on block
+// device object
+func BuilderForObject(BlockDeviceClaim *BlockDeviceClaim) *Builder {
+	return &Builder{
+		BDC: BlockDeviceClaim,
+	}
+}
+
+// BuilderForAPIObject returns an instance of the Builder object based on block
+// device claim api object.
+func BuilderForAPIObject(bdc *ndmapisv1alpha1.BlockDeviceClaim) *Builder {
+	return &Builder{
+		BDC: &BlockDeviceClaim{bdc},
+	}
+}
+
+// Build returns the BlockDeviceClaim instance
+func (b *Builder) Build() (*BlockDeviceClaim, error) {
+	if len(b.errs) > 0 {
+		return nil, errors.Errorf("%+v", b.errs)
+	}
+	return b.BDC, nil
+}
+
+// WithName sets the Name field of PVC with provided value.
+func (b *Builder) WithName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build BDC object: missing BDC name"))
+		return b
+	}
+	b.BDC.Object.Name = name
+	return b
+}
+
+// WithNamespace sets the Namespace field of PVC provided arguments
+func (b *Builder) WithNamespace(namespace string) *Builder {
+	if len(namespace) == 0 {
+		namespace = "default"
```

Comment:
Let us append error if namespace is provided as blank.
Caller may not invoke this method at all, if it wishes.